kyo-tag/shared/src/main/scala/kyo/tags.scala

```diff
@@ -2,110 +2,132 @@ package kyo
+
     import internal.*
 
     inline given apply[T]: Tag[T] = ${ tagImpl[T] }
 
     extension [T](t1: Tag[T])
+        def erased: Tag[Any] = t1.asInstanceOf[Tag[Any]]
 
-        def show: String =
-            val decoded = t1.tpe.drop(2).takeWhile(_ != ';')
-            fromCompact.getOrElse(decoded, decoded)
-
-        infix def <:<[U](t2: Tag[U]): Boolean =
-            t1 =:= t2 || isSubtype(t1, t2)
-
-        infix def =:=[U](t2: Tag[U]): Boolean =
-            (t1.tpe eq t2.tpe) || t1.tpe == t2.tpe
-
-        infix def =!=[U](t2: Tag[U]): Boolean =
-            (t1.tpe ne t2.tpe) || t1.tpe != t2.tpe
+    extension [T](t1: Full[T])
 
-        infix def >:>[U](t2: Tag[U]): Boolean =
-            t1 =:= t2 || isSubtype(t2, t1)
-
-    end extension
+        def show: String =
+            t1 match
+                case t1: String =>
+                    val decoded = t1.drop(2).takeWhile(_ != ';')
+                    fromCompact.getOrElse(decoded, decoded)
+                case t1: Set[T] =>
+                    t1.show
 
-    case class Union[T](tags: Seq[Tag[?]]) extends AnyVal
+        infix def <:<[U](t2: Full[U]): Boolean =
+            t1 match
+                case t1: String =>
+                    t2 match
+                        case t2: String => isSubtype(t1, t2)
+                        case t2: Set[?] => t2 >:> t1
+                case t1: Set[T] =>
+                    t1 <:< t2
 
-    object Union:
+        infix def =:=[U](t2: Full[U]): Boolean =
+            (t1 eq t2) || t1 == t2
 
-        inline given apply[T]: Union[T] = ${ unionImpl[T] }
+        infix def =!=[U](t2: Full[U]): Boolean =
+            !(t1 =:= t2)
 
-        extension [T](t1: Union[T])
-            def show: String =
-                t1.tags.map(_.show).mkString(" | ")
+        infix def >:>[U](t2: Full[U]): Boolean =
+            t2 <:< t1
 
-            infix def <:<[U](t2: Tag[U]): Boolean =
-                t1.tags.forall(_ <:< t2)
+        def erased: Full[Any] = t1.asInstanceOf[Full[Any]]
 
-            infix def =:=[U](t2: Tag[U]): Boolean =
-                t1.tags.forall(t2 =:= _)
+    end extension
 
-            infix def =!=[U](t2: Tag[U]): Boolean =
-                t1.tags.exists(t2 =!= _)
+    sealed trait Set[T]:
+        def show: String
+        infix def <:<[U](t2: Full[U]): Boolean
+        infix def =:=[U](t2: Full[U]): Boolean
+        infix def =!=[U](t2: Full[U]): Boolean =
+            !(this =:= t2)
+        infix def >:>[U](t2: Full[U]): Boolean
+        def erased: Set[Any] = this.asInstanceOf[Set[Any]]
+    end Set
 
-            infix def >:>[U](t2: Tag[U]): Boolean =
-                t1.tags.exists(t2 <:< _)
+    object Set:
+        inline given apply[T]: Set[T] = ${ setImpl[T] }
 
-            infix def <:<(t2: Union[?]): Boolean =
-                t1.tags.forall(tag1 => t2.tags.exists(tag1 <:< _))
+    case class Union[T](tags: Seq[Tag[Any]]) extends Set[T]:
```

Comment:
I believe this can extend AnyVal

Reply:
Nice! I thought it wouldn't be possible but I just needed to make `Set` a [universal trait](https://docs.scala-lang.org/overviews/core/value-classes.html).